File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.cpp
#include <algorithm>
#include <iostream>
#include <map>
#include <queue>
#include <vector>

using namespace std;

vector<int> solution(vector<int> emergency) {
    vector<int> answer = emergency;
    vector<int> result;
    map<int, int> m;

    sort(emergency.begin(), emergency.end(), greater<int>());

    for (int i = 0; i < emergency.size(); i++) {
        m[emergency[i]] = i + 1;
    }

    for (int i = 0; i < answer.size(); i++) {
        result.push_back(m[answer[i]]);
    }

    return result;
}

File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.swift
import Foundation

func solution(_ emergency:[Int]) -> [Int] {
    
    var result: [Int:Int] = [:]
    var index = 1
    
    emergency.sorted(by: >).forEach {
        result.updateValue(index, forKey: $0)
        index += 1
    }
    
    return emergency.map { result[$0]! }
}